src/structurizr/api/structurizr_client.py

```diff
@@ -109,14 +114,36 @@ def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
-        """Exit a context by unlocking the corresponding remote workspace."""
+        """Exit a context by unlocking the corresponding remote workspace.
+
+        Note: this method is deprecated in favour of lock(), and will be removed in a
+        future relesae.
+        """
         is_successful = self.unlock_workspace()
         self._client.close()
         if exc_type is None and not is_successful:
             raise StructurizrClientException(
                 f"Failed to unlock the Structurizr workspace {self.workspace_id}."
             )
 
+    @contextmanager
+    def lock(self):
```

Comment:
Can't decide whether I'd like this name to be `lock` or `lock_context`... 🤔 I suppose `lock` is consistent with other class interfaces.

Reply:
or possibly `auto_lock`?

Reply:
Not sure I like `auto_lock`, as it doesn't make it automatic!  We could go for `workspace_lock()` as that reads quite well, but that might get confusing versus the `lock_workspace()` method.  If we do go for `workspace_lock()` then I'd be tempted to make that a property rather than a method:

    with client.workspace_lock:
        # Do some stuff

vs

    with client.lock():
        # Do some stuff

Reply:
Let's keep it as is then 🙂